frontend/pages/index.vue

```diff
@@ -1016,51 +999,37 @@ export default {
           }
         })
     },
+    showArtsClusterModal(placenames, coordinates) {
+      const mapboxgl = require('mapbox-gl')
 
-    showArtsClusterModal(feature, latLng, map) {
-      const clusterId = feature.properties.cluster_id
-      map
-        .getSource('arts1')
-        .getClusterLeaves(
-          clusterId,
-          feature.properties.point_count,
-          0,
-          function(err, aFeatures) {
-            if (err) {
-              console.log('Error', err)
-            }
-            const html = aFeatures.reduce(function(ach, feature) {
-              const props = feature.properties
-              return ach + renderArtDetail(props)
-            }, '')
-            const mapboxgl = require('mapbox-gl')
-            new mapboxgl.Popup({
-              className: 'artPopUp'
-            })
-              .setLngLat(latLng)
-              .setHTML(
-                `<div class='popup-inner'>
-                    <h4>Art Here:</h4>
-
-                    ${html}
-                    <!-- TODO scroll indicator -->
-                    <div class="scroll-indicator">
-                        <i class="fas fa-angle-down float"></i>
-                    </div>
-
-                    </div>`
-              )
-              .addTo(map)
-          }
-        )
+      const lnglat = new mapboxgl.LngLat(coordinates[0], coordinates[1])
+
+      const popup = new mapboxgl.Popup({
+        className: 'cluster-modal'
+      })
+        .setLngLat(lnglat)
+        .setHTML(`<div id="art-cluster-container"></div>`)
+        .addTo(this.map)
+
+      const clusterModal = new ArtsClusterModalClass({
+        propsData: {
+          placenames
+        },
+        methods: {
+          goToPlacename: this.goToPlacename
+        }
+      })
+      clusterModal.$mount('#art-cluster-container')
```

Comment:
This is actually a good and cleaner implementation. I have not thought of using a component for mapboxql popup.